xwiki-pro-macros-api/src/main/java/com/xwiki/macros/showhideif/macro/ShowHideIfMacroParameters.java

```diff
@@ -147,4 +168,20 @@ public void setGroups(GroupReferenceList groups)
     {
         this.groups = groups;
     }
+
+    /**
+     * @return the display type which should match.
```

Comment:
the description is a bit vague. From what I understand, this parameter should add some additional displaying constraints? If it's NONE, this parameter is ignored and the macro will display according to the other params. DEFAULT will display in all the context actions (view/edit/etc) except `export` and PRINTABLE will display only when the context action is export?

You should also add the @PropertyDescription annotation. Ideally, some translations should be added.

Reply:
This is how you create translations for the macro parameters https://www.xwiki.org/xwiki/bin/view/Documentation/DevGuide/Tutorials/WritingMacros/WikiMacroTutorial/#HTranslations

The current implementation is alright. We can improve it in the future if needed be.